src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -136,7 +135,7 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
 
     val world2 = (refundGasFn andThen payMinerForGasFn andThen deleteAccountsFn andThen persistStateFn)(resultWithErrorHandling.world)
 
-    TxResult(world2, gasUsed, resultWithErrorHandling.logs)
+    TxResult(world2, gasLimit - gasRefund, resultWithErrorHandling.logs)
```

Comment:
@rtkaczyk I think i've addressed your comments. I think the only one left is the one that states:

> this comment is inaccurate  - it's the whole sum. 

Can you rephrase pls?